File: MyApp19-master/app/src/main/java/com/example/app19/MainActivity.java
package com.example.app19;

import static androidx.constraintlayout.helper.widget.MotionEffect.TAG;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.util.Patterns;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MainActivity extends AppCompatActivity {
    Button bt1,bt2;
    EditText em1,pass1;
    private FirebaseAuth firebaseAuth;
    private FirebaseUser user;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        bt1 = findViewById(R.id.button);
        bt2 = findViewById(R.id.button3);
        em1=findViewById(R.id.editTextTextEmailAddress2);
        pass1=findViewById(R.id.editTextTextPassword2);

        firebaseAuth = FirebaseAuth.getInstance();
        user = firebaseAuth.getInstance().getCurrentUser();
        if(user!=null){
            Intent i = new Intent(MainActivity.this,MainActivity3.class);
            startActivity(i);
        }
        bt1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String mail,pass;
                mail=em1.getText().toString();
                pass=pass1.getText().toString();
                boolean VV = Valid(mail,pass);
                if(VV) {
                    firebaseAuth.signInWithEmailAndPassword(mail, pass).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                        @Override
                        public void onComplete(@NonNull Task<AuthResult> task) {
                            if (task.isSuccessful()) {
                                Toast.makeText(MainActivity.this, "User logged in successful", Toast.LENGTH_LONG).show();
                                Intent i = new Intent(MainActivity.this, MainActivity3.class);
                                startActivity(i);
                            } else {
                                Toast.makeText(MainActivity.this, "User logged in unsuccessful", Toast.LENGTH_LONG).show();
                            }
                        }
                    });
                }
            }
        });
        bt2.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent i = new Intent(MainActivity.this,MainActivity2.class);
                startActivity(i);
                finish();
            }
        });
    }
    private  boolean Valid(String email,String pass){
        if(!Patterns.EMAIL_ADDRESS.matcher(email).matches()){
            Toast.makeText(MainActivity.this, "The mail is incorrect", Toast.LENGTH_LONG).show();
            return false;
        }else if (TextUtils.isEmpty(pass)){
            Toast.makeText(MainActivity.this, "Password is empty", Toast.LENGTH_LONG).show();
            return false;
        }else if(pass.length()<4) {
            Toast.makeText(MainActivity.this, "Password less than 4 characters ", Toast.LENGTH_LONG).show();
            return false;
        }else{
            return true;
        }
    }
}

File: MyApp20Kotlin-master/app/src/main/java/com/example/app20kotlin/MainActivity.kt
package com.example.app20kotlin

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Patterns
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser
import com.google.firebase.firestore.FirebaseFirestore

class MainActivity : AppCompatActivity() {
    private var ed1: EditText?=null
    private  var ed2:EditText?=null
    private var btn1: Button?=null
    private  var btn2:Button?=null
    var uid: String?=null
    private val auth = FirebaseAuth.getInstance()
    private var user: FirebaseUser?=null
    private val db: FirebaseFirestore = FirebaseFirestore.getInstance()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var userRef = db.collection("users")
        ed1 = findViewById(R.id.editText1)
        ed2 = findViewById(R.id.editText2)
        btn1 = findViewById(R.id.button1)
        btn2 = findViewById(R.id.button2)

        btn1!!.setOnClickListener{
            val VV: Boolean = Valid(ed1!!.text.toString(), ed2!!.text.toString())
            if (VV) {
                auth.signInWithEmailAndPassword(ed1!!.text.toString(), ed2!!.text.toString())
                    .addOnCompleteListener(this) { task ->
                        if (task.isSuccessful) {
                            user = task.result.user!!
                            uid = user!!.uid
                            val i = Intent(this, MainActivity3::class.java)
                            i.putExtra("uid", uid)
                            startActivity(i)
                        } else {
                            Toast.makeText(
                                this,
                                "User logged in unsuccessful",
                                Toast.LENGTH_LONG
                            ).show()
                        }
                    }
            }
        }
        btn2!!.setOnClickListener {
            val i = Intent(this, MainActivity2::class.java)
            startActivity(i)
        }
    }
    private fun Valid(email: String, pass: String): Boolean {
        return if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            Toast.makeText(this, "The mail is incorrect", Toast.LENGTH_LONG).show()
            false
        } else if (TextUtils.isEmpty(pass)) {
            Toast.makeText(this, "Password is empty", Toast.LENGTH_LONG).show()
            false
        } else if (pass.length < 4) {
            Toast.makeText(this, "Password less than 4 characters ", Toast.LENGTH_LONG)
                .show()
            false
        } else {
            true
        }
    }
}